### DISCUSSION TITLE
Set elasticsearch options dynamically

### DISCUSSION THREAD

**Discussion by poppelaars** [wallneradam/esorm#19] [Q&A] [answered]
Hi,

When using the esproxy I am able to call functions from the elasticsearch library.

```python
from esorm import es

user = await es.options(
            headers={
                "es-security-runas-user": username,
            },
            basic_auth=(settings.elasticsearch_username, settings.elasticsearch_password)
        ).security.authenticate()
```

Is it possible "out-of-the-box" to set elasticsearch options dynamically, such as a header, dynamically when using an esorm model? For example inside:

```python
from esorm import ESModel

# Here the model have automatically generated id
class User(ESModel):
    name: str
    age: int


async def create_user():
    # Create a new user
    user = User(name='John Doe', age=25)
    # Save the user to ElasticSearch
    new_user_id = await user.save()
    print(new_user_id)
```

**Comment by wallneradam** [comment, marked as answer]
I've never used this feature of ES. So you would like to have different headers for the communications of a specific model?

If you check the source code, there is a `call` method of the `ESModel`. You could easily override it. Then every ES call will use your headers.
